src/services/user.js

```diff
@@ -315,6 +315,46 @@ const getMemberships = async id => {
   memberships = await http.get(`memberships/student/${id}`);
   return memberships;
 };
+
+const getTranscriptInfo = async id => {
+  let transcriptInfo = await getMemberships(id);
+  console.log(transcriptInfo);
+  transcriptInfo.activities = [0];
+  let k = 0;
+  let second = 1;
+  let i = 0;
+  for (let first = 0; first < transcriptInfo.length; first++) {
+    if (transcriptInfo[first].SessionCode !== transcriptInfo[second].SessionCode) {
+      if (k === first) {
+        let obj = [];
+        obj.push(transcriptInfo[k].ActivityDescription);
+        transcriptInfo.activities.push(obj);
+        k++;
+      } else {
+        let obj = [];
+        while (k !== first) {
+          obj.push(transcriptInfo[k].ActivityDescription);
+          k++;
+        }
+        transcriptInfo.activities.push(obj);
+      }
+    }
+    first++;
```

Comment:
It looks like "first" will be incremented twice per iteration, moving ahead twice as fast as "second".  Is that what's intended?